src/commands.c

```diff
@@ -1412,12 +1548,116 @@ int yh_com_open_session(yubihsm_context *ctx, Argument *argv, cmd_format fmt) {
   return 0;
 }
 
+// NOTE(adma): Open a session with a connector using an Asymmetric
+// Authentication Key argc = 2 arg 0: w:authkey arg 1: i:password
+int yh_com_open_session_asym(yubihsm_context *ctx, Argument *argv,
+                             cmd_format in_fmt, cmd_format fmt) {
+
+  UNUSED(fmt);
+
+  yh_session *ses = NULL;
+  uint8_t session_id = 0;
+
+  if (ctx->connector == NULL) {
+    fprintf(stderr, "Not connected\n");
+    return -1;
+  }
+
+  uint16_t authkey = argv[0].w;
+  uint8_t privkey[32];
+  yh_rc yrc;
+
+  if (in_fmt == fmt_password) {
```

Comment:
I'm not convinced about the different behaviour depending on in_fmt since nothing else has that, on the other hand I don't have a better suggestion (so feel free to ignore this comment).

Reply:
I added this to be able to get the format needed in config files

Reply:
Also please note that I fixed yubihsm-shell to allow a command line option to set this and then enter interactive mode. Before the informat and outformat command line options were not processes unless an action was also specified.